src/gui/sdlmain.cpp

```diff
@@ -4646,13 +4673,24 @@ void restart_dosbox(std::vector<std::string> &parameters) {
 			current_directory,
 			&startup_info,
 			&process_information)) {
-		E_Exit("Restarting failed");
+		LOG_ERR("Restart failed: CreateProcess failed");
 	}
-	ExitProcess(ERROR_SUCCESS);
 #else
-	if (execvp(newargs[0], newargs) == -1) {
-		E_Exit("Restarting failed");
+	int ret = fork();
+	switch (ret) {
+		case -1:
+			LOG_ERR("Restart failed: fork failed: %s", strerror(errno));
+			break;
+		case 0:
+			// Newly created child process immediately executes a new instance of DOSBox.
+			// It is not safe for a child process in a multi-threaded program to do much else.
+			if (execvp(newargs[0], newargs) == -1) {
+				E_Exit("Restart failed: execvp failed: %s", strerror(errno));
```

Comment:
Any reason for the hard exit? Wouldn't `LOG_ERR` be better?

Reply:
This is in the child process. It crashes if you try to do a normal exit. The parent process will do a normal exit regardless if we've gotten this far. If the child fails `execvp()`, we have to kill it.